crates/recursor/src/recursor_dns_handle.rs

```diff
@@ -27,9 +26,10 @@ use crate::{
 /// Set of nameservers by the zone name
 type NameServerCache<P> = LruCache<Name, RecursorPool<P>>;
 
+#[derive(Clone)]
```

Comment:
Why is this necessary? Are the `RecursorPool` and `DnsLru` cheap to clone? Why not wrap the entire `RecursorDnsHandle` in an `Arc`?

Reply:
> Why is this necessary?

`DnsHandle` is a subtrait of `Clone` and we need to implement `DnsHandle` to put this type inside `DnssecDnsHandle<_>`

> Are the RecursorPool and DnsLru cheap to clone?

`RecursorPool` has several `Arc` fields and  `DnsLru` is esentially an `Arc<Mutex<_>>` so, yes

> Why not wrap the entire RecursorDnsHandle in an Arc?

doable but that would result in a nested `Arc` which ... I guess it's fine? I don't think either way has much impact on performance, either CPU or memory usage.

Reply:
I guess it's fine, no need to change it on my account.